cli/injector/main.go

```diff
@@ -329,6 +339,50 @@ func injectAndWait(cmd *cobra.Command, args []string) {
 		}
 	}
 
+	if len(pulsingInjectors) > 0 {
+		isInjected := true
+		lastOperationTime := time.Now()
+
+		for { // This loop will inject, wait, clean, wait until a signal is received
+			// Inject all the pulsing disruptions
+			if time.Since(lastOperationTime) >= pulseDormantDuration && !isInjected {
+				for _, inj := range pulsingInjectors {
+					if err := inj.Inject(); err != nil {
+						log.Errorw("pulsing disruption injection failed", "error", err)
+					}
+				}
+
+				isInjected = !isInjected
+				lastOperationTime = time.Now()
+
+				log.Info("pulsing disruption(s) are active")
+			} else if time.Since(lastOperationTime) >= pulseActiveDuration && isInjected { // Clean all the pulsing disruptions
+				for _, inj := range pulsingInjectors {
+					// start cleanup which is retried up to 3 times using an exponential backoff algorithm
+					if err := backoff.RetryNotify(inj.Clean, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 3), retryNotifyHandler); err != nil {
```

Comment:
So far in testing I haven't run into an issue where we delete the chaos pod while it's dormant (And thus we'd be cleaning a cleaned target). 

I am a little concerned that we do these 3 retries for each (Inj and Clean), and after that just log an Error and move on. Should we maybe fatal instead? @Devatoria

Reply:
I agree, we should probably exit here to catch the error earlier. What about injection though? Should we do the same thing?